python/adaptivity.py.in

```diff
@@ -459,23 +455,24 @@ def adapt(metric, bfaces=None, bfaces_func=None, colinearity_tol = 30, debugon=T
         or not element.family() == "Lagrange" \
         or not element.degree() == 1:
     raise InvalidArgumentException("Require 2D P1 function space for metric tensor field")
-  
+
   gdim = element.cell().geometric_dimension()
 
-  nodes = array(range(0,mesh.num_vertices()),dtype=numpy.intc) 
+  nodes = array(range(0,mesh.num_vertices()),dtype=numpy.intc)
   cells_array = mesh.cells().astype(numpy.intc)
   coords = mesh.coordinates()
   # create boundary mesh and associated list of co-linear edges
   if bfaces is None:
       mesh.init(1,2)
-      bfaces = set([])
-      [bfaces.add(f) for f in facets(mesh) if f.exterior()]
-      bfaces_list = list(bfaces)
+      bfaces_list = []
+      # set((bfaces.add(f) for f in facets(mesh) if f.exterior()))
+      [bfaces_list.append(f) for f in facets(mesh) if f.exterior()]
+#      bfaces = set(bfaces_list)
```

Comment:
Can you please remove the commented lines if they are not needed?

I wonder why I used set... odd. After this I will merge. Thanks a lot!